From 89f0507145bc3d0c03bce04f3cf831eda4c4ccd6 Mon Sep 17 00:00:00 2001
From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Tue, 31 Mar 2015 12:53:22 +0200
Subject: [PATCH] Revert "can: at91_can: add missing prepare and unprepare of
 the clock"

This reverts commit 51562cd4de104521223f8e4e9cbe04bef401a79f.

The driver fails to compile due to:

drivers/net/can/at91_can.c:1118:2: error: implicit declaration of function 'clk_prepare_enable' [-Werror=implicit-function-declaration]
drivers/net/can/at91_can.c:1144:2: error: implicit declaration of function 'clk_disable_unprepare' [-Werror=implicit-function-declaration]

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 drivers/net/can/at91_can.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/drivers/net/can/at91_can.c b/drivers/net/can/at91_can.c
index a42d75c..99f889e 100644
--- a/drivers/net/can/at91_can.c
+++ b/drivers/net/can/at91_can.c
@@ -1116,9 +1116,7 @@ static int at91_open(struct net_device *dev)
 	struct at91_priv *priv = netdev_priv(dev);
 	int err;
 
-	err = clk_prepare_enable(priv->clk);
-	if (err)
-		return err;
+	clk_enable(priv->clk);
 
 	/* check or determine and set bittime */
 	err = open_candev(dev);
@@ -1142,7 +1140,7 @@ static int at91_open(struct net_device *dev)
  out_close:
 	close_candev(dev);
  out:
-	clk_disable_unprepare(priv->clk);
+	clk_disable(priv->clk);
 
 	return err;
 }
@@ -1159,7 +1157,7 @@ static int at91_close(struct net_device *dev)
 	at91_chip_stop(dev, CAN_STATE_STOPPED);
 
 	free_irq(dev->irq, dev);
-	clk_disable_unprepare(priv->clk);
+	clk_disable(priv->clk);
 
 	close_candev(dev);
 
-- 
1.8.3.1

